clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -158,6 +158,22 @@ public KafkaPrincipal principal() {
         return authenticator.principal();
     }
 
+    /**
+     * Returns the ClientSoftwareName extracted from the ApiVersionsRequest which may
+     * have been received by the Authenticator or an empty String.
+     */
+    public String clientSoftwareName() {
+        return authenticator.clientSoftwareName();
+    }
+
+    /**
+     * Returns the ClientSoftwareVersion extracted from the ApiVersionsRequest which may
+     * have been received by the Authenticator or an empty String.
```

Comment:
We should spell out that if the string is empty, it's because the remote end didn't send this information